homeassistant/components/tessie/entity.py

```diff
@@ -43,3 +48,40 @@ def __init__(
     def _value(self) -> Any:
         """Return value from coordinator data."""
         return self.coordinator.data[self.key]
+
+    def get(self, key: str | None = None, default: Any | None = None) -> Any:
+        """Return a specific value from coordinator data."""
+        return self.coordinator.data.get(key or self.key, default)
+
+    async def run(
+        self, func: Callable[..., Awaitable[dict[str, bool]]], **kargs: Any
+    ) -> bool:
+        """Run a tessie_api function and handle exceptions."""
+        try:
+            response = await func(
+                session=self.coordinator.session,
+                vin=self.vin,
+                api_key=self.coordinator.api_key,
+                **kargs,
+            )
+        except ClientResponseError as e:
+            if e.status == HTTPStatus.INTERNAL_SERVER_ERROR:
+                # Create issue for Virtual Key setup
+                ir.async_create_issue(
+                    self.hass,
+                    DOMAIN,
+                    "virtual_key",
+                    is_fixable=True,
+                    is_persistent=False,
+                    learn_more_url="https://help.tessie.com/article/117-virtual-key",
+                    severity=ir.IssueSeverity.ERROR,
+                    translation_key="virtual_key",
+                )
```

Comment:
Please move the issue creation to another PR

You'll also likely want to dismiss the issue on the success path when they fix it

Reply:
This has been removed.